packages/airnode-validator/src/config/config.ts

```diff
@@ -138,10 +139,35 @@ export const apiCredentialsSchema = baseApiCredentialsSchema.extend({
   oisTitle: z.string(),
 });
 
-export const configSchema = z.object({
-  chains: z.array(chainConfigSchema),
-  nodeSettings: nodeSettingsSchema,
-  ois: z.array(oisSchema),
-  triggers: triggersSchema,
-  apiCredentials: z.array(apiCredentialsSchema),
-});
+const validateSecuritySchemesReferences: SuperRefinement<{
+  ois: OIS[];
+  apiCredentials: ApiCredentials[];
+}> = (config, ctx) => {
+  config.ois.forEach((ois, index) => {
+    Object.keys(ois.apiSpecifications.security).forEach((enabledSecuritySchemeName) => {
+      const enabledSecurityScheme = ois.apiSpecifications.components.securitySchemes[enabledSecuritySchemeName];
+      if (enabledSecurityScheme && ['apiKey', 'http'].includes(enabledSecurityScheme.type)) {
+        const securitySchemeApiCredentials = config.apiCredentials.find(
+          (apiCredentials) => apiCredentials.securitySchemeName === enabledSecuritySchemeName
+        );
+        if (!securitySchemeApiCredentials || !securitySchemeApiCredentials.securitySchemeValue) {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: `The security scheme is enabled but no credentials are provided`,
+            path: ['ois', index, 'apiSpecifications', 'security', enabledSecuritySchemeName],
```

Comment:
The value in the path should be index not a name.

```suggestion
            path: ['ois', index, 'apiSpecifications', 'security', schemeIndex],
```

Reply:
🤔 sure about this? ois.apiSpecifications.security is not an array but an object where securitySchemeNames are the keys of it's properties. I can easily add the index but I'm wondering if it would be easier to determine where the error is if I just use the property key in the path. WDYT?

Reply:
🤦 Sorry, the security is a an object. I got confused somewhere  and thought it was an array. We definitely want a name not index.